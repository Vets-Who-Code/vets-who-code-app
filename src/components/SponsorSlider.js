import React from 'react';
import Slider from 'react-slick';

import pluralsightPNG from '../images/supporters/ps.png';
import basecampPNG from '../images/supporters/basecamp-logo.png';
import replItPNG from '../images/supporters/repl.it.png';
import slackPNG from '../images/supporters/slack.png';
import doPNG from '../images/supporters/DO.png';
import icPNG from '../images/supporters/IC.png';
import honeyBadgerPNG from '../images/supporters/honeybadger.png';

const settings = {
  dots: true,
  infinite: true,
  speed: 500,
  slidesToShow: 7,
  className: 'slider',
  centerMode: true,
  centerPadding: 0,
  slidesToScroll: 1,
  autoplay: true,
  autoplaySpeed: 3000,
  responsive: [
    {
      breakpoint: 990,
      settings: {
        slidesToShow: 4,
        slidesToScroll: 1,
      },
    },
    {
      breakpoint: 480,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 1,
      },
    },
  ],
};

const SponsorSlider = () => (
  <Slider {...settings}>
    <div>
      <a href="#">
        <img src={pluralsightPNG} alt="Pluralsight" height="60" width="60" />
      </a>
    </div>
    <div>
      <a href="#">
        <img src={basecampPNG} alt="Basecamp" height="60" width="60" />
      </a>
    </div>
    <div>
      <a href="#">
        <img src={replItPNG} alt="repl.it" height="60" width="60" />
      </a>
    </div>
    <div>
      <a href="#">
        <img src={slackPNG} alt="Slack" height="60" width="60" />
      </a>
    </div>
    <div>
      <a href="#">
        <img src={doPNG} alt="Digital Ocean" height="60" width="60" />
      </a>
    </div>
    <div>
      <a href="#">
        <img src={icPNG} alt="Interview Cake" height="60" width="60" />
      </a>
    </div>
    <div>
      <a href="#">
        <img src={honeyBadgerPNG} alt="Honeybadger" height="60" width="60" />
      </a>
    </div>
  </Slider>
);

export default SponsorSlider;
