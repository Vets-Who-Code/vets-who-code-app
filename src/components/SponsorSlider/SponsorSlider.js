import { useState, useEffect, useContext } from 'react'
import Image from 'next/image'
import Carousel from 'nuka-carousel'
import { ThemeContext } from '../../store/ThemeProvider'
import { FaSlack, FaGoogle, FaGithub } from 'react-icons/fa'
import { SiReplit } from 'react-icons/si'
import { Contentful, Fem, Vercel } from '../../icons'

const baseSettlings = {
  autoplay: true,
  enableKeyboardControls: true,
  pauseOnHover: true,
  speed: 500,
  wrapAround: true,
  withoutControls: true,
}

const alignmentStyles = {
  position: 'relative',
  overflow: 'hidden',
  height: 60,
  width: 60,
  margin: '0px auto',
}

const elements = [
  {
    href: 'https://www.google.com/',
    label: 'Link to Google',
    title: 'Google',
    // /images is in the public folder
    lightElement: '/images/supporters/google.png',
    darkElement: {
      component: FaGoogle,
      props: {
        size: 60,
        color: '#ffffff',
      },
    },
  },
  {
    href: 'https://github.com/',
    label: 'Link to Github',
    title: 'GitHub',
    // /images is in the public folder
    lightElement: '/images/supporters/github.png',
    darkElement: {
      component: FaGithub,
      props: {
        size: 60,
        color: '#ffffff',
      },
    },
  },
  {
    href: 'https://repl.it',
    label: 'Link to Replit ',
    title: 'Repl.it',
    // /images is in the public folder
    lightElement: '/images/supporters/repl.it.png',
    darkElement: {
      component: SiReplit,
      props: {
        size: 60,
        color: '#ffffff',
      },
    },
  },
  {
    href: 'https://slack.com/',
    label: 'Link to Slack',
    title: 'Slack',
    // /images is in the public folder
    lightElement: '/images/supporters/slack.png',
    darkElement: {
      component: FaSlack,
      props: {
        size: 60,
        color: '#ffffff',
      },
    },
  },
  {
    href: 'https://frontendmasters.com/',
    label: 'Link to Front End Masters',
    title: 'Front End Masters',
    // /images is in the public folder
    lightElement: '/images/supporters/fem.png',
    darkElement: {
      component: Fem,
      props: {
        size: 60,
        color: '#ffffff',
      },
    },
  },
  {
    href: 'https://www.contentful.com/',
    label: 'Link to Contentful',
    title: 'Contentful',
    // /images is in the public folder
    lightElement: '/images/supporters/contentful.png',
    darkElement: {
      component: Contentful,
      props: {
        size: 60,
        color: '#0191f40',
      },
    },
  },
  {
    href: 'https://vercel.com?utm_source=vetswhocode.io&utm_campaign=oss',
    label: 'Link to Vercel',
    title: 'Vercel',
    // /image is in the public folder
    lightElement: '/images/supporters/vercel.png',
    darkElement: {
      component: Vercel,
      props: {
        size: 60,
        color: '#ffffff',
      },
    },
  },
]

function SponsorSlider() {
  const { colorMode } = useContext(ThemeContext)

  const [viewport, setViewport] = useState(800)

  function updateWindowDimensions() {
    setViewport(window.innerWidth)
  }

  useEffect(() => {
    window.addEventListener('resize', updateWindowDimensions)
    return () => window.removeEventListener('resize', updateWindowDimensions)
  }, [])

  const isMobile = Boolean(viewport < 800)

  const lightElements = () => {
    return elements.map(data => {
      return (
        <div key={data.title} style={{ textAlign: 'center' }}>
          <a
            href={data.href}
            aria-label={data.label}
            title={data.title}
            target="_blank"
            rel="noopener noreferrer"
          >
            <Image
              src={data.lightElement}
              alt={data.title}
              height={alignmentStyles.height}
              width={alignmentStyles.width}
            />
          </a>
        </div>
      )
    })
  }

  const darkElements = () => {
    return elements.map(data => {
      const Icon = data.darkElement.component
      const IconProps = data.darkElement.props
      return (
        <div key={data.title} style={{ textAlign: 'center' }}>
          <a
            href={data.href}
            aria-label={data.label}
            title={data.title}
            target="_blank"
            rel="noopener noreferrer"
          >
            <Icon {...IconProps} />
          </a>
        </div>
      )
    })
  }

  return (
    <Carousel
      {...baseSettlings}
      slidesToShow={isMobile ? 4 : 7}
      transitionMode={isMobile ? 'scroll' : 'fade'}
    >
      {colorMode === 'light' ? lightElements() : darkElements()}
    </Carousel>
  )
}

export default SponsorSlider
