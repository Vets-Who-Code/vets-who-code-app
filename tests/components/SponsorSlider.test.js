import { render } from '@testing-library/react'
import SponsorSlider from '@/components/SponsorSlider'

describe('<SponsorSlider />', () => {
  test('should render correctly', () => {
    const { container } = render(<SponsorSlider />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
