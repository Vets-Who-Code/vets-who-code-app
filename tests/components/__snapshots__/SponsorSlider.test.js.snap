// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SponsorSlider /> should render correctly 1`] = `
<div
  class="slider"
  style="box-sizing: border-box; display: block; height: inherit; position: relative; width: 100%;"
  tabindex="0"
>
  <div
    class="slider-frame"
    style="box-sizing: border-box; display: block; height: 100%; margin: 0px; overflow: hidden; padding: 0px; position: relative; transform: translate3d(0, 0, 0);"
  >
    <ul
      class="slider-list"
      style="box-sizing: border-box; cursor: pointer; display: block; height: 0px; margin: 0px 0px; padding: 0px; width: 0px;"
    >
      <li
        class="slider-slide slide-visible"
        style="box-sizing: border-box; display: block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; opacity: 1; position: absolute; top: 0px; vertical-align: top; visibility: inherit; width: 0px;"
        tabindex="-1"
      >
        <a
          aria-label="Link to Google"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="Google"
        >
          static-query
        </a>
      </li>
      <li
        class="slider-slide slide-visible"
        style="box-sizing: border-box; display: block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; opacity: 1; position: absolute; top: 0px; vertical-align: top; visibility: inherit; width: 0px;"
        tabindex="-1"
      >
        <a
          aria-label="Link to Github"
          href="https://github.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="GitHub"
        >
          static-query
        </a>
      </li>
      <li
        class="slider-slide slide-visible"
        style="box-sizing: border-box; display: block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; opacity: 1; position: absolute; top: 0px; vertical-align: top; visibility: inherit; width: 0px;"
        tabindex="-1"
      >
        <a
          aria-label="Link to Replit "
          href="https://repl.it"
          rel="noopener noreferrer"
          target="_blank"
          title="Repl.it"
        >
          static-query
        </a>
      </li>
      <li
        class="slider-slide slide-visible"
        style="box-sizing: border-box; display: block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; opacity: 1; position: absolute; top: 0px; vertical-align: top; visibility: inherit; width: 0px;"
        tabindex="-1"
      >
        <a
          aria-label="Link to Slack"
          href="https://slack.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="Slack"
        >
          static-query
        </a>
      </li>
      <li
        class="slider-slide slide-visible"
        style="box-sizing: border-box; display: block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; opacity: 1; position: absolute; top: 0px; vertical-align: top; visibility: inherit; width: 0px;"
        tabindex="-1"
      >
        <a
          aria-label="Link to Front End Masters"
          href="https://frontendmasters.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="Front End Masters"
        >
          static-query
        </a>
      </li>
      <li
        class="slider-slide slide-visible"
        style="box-sizing: border-box; display: block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; opacity: 1; position: absolute; top: 0px; vertical-align: top; visibility: inherit; width: 0px;"
        tabindex="-1"
      >
        <a
          aria-label="Link to Contentful"
          href="https://www.contentful.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="Contentful"
        >
          static-query
        </a>
      </li>
    </ul>
  </div>
  <style
    type="text/css"
  >
    .slider-slide &gt; img { width: 100%; display: block; }
          .slider-slide &gt; img:focus { margin: auto; }
  </style>
</div>
`;
